scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -131,42 +137,106 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
      */
     fun getDetails() = ScannerDetails(scannerName, getVersion(), getConfiguration())
 
-    override fun scanPackages(packages: List<Package>, outputDirectory: File, downloadDirectory: File):
+    // `newFixedThreadPoolContext` is obsolete and will be replaced with a new API, see:
+    // https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/new-fixed-thread-pool-context.html
+    @ObsoleteCoroutinesApi
+    override suspend fun scanPackages(packages: List<Package>, outputDirectory: File, downloadDirectory: File):
             Map<Package, List<ScanResult>> {
         val scannerDetails = getDetails()
 
-        return packages.withIndex().associate { (index, pkg) ->
-            val result = try {
-                log.info { "Starting scan of '${pkg.id.toCoordinates()}' (${index + 1}/${packages.size})." }
-
-                scanPackage(scannerDetails, pkg, outputDirectory, downloadDirectory).map {
-                    // Remove the now unneeded reference to rawResult here to allow garbage collection to clean it up.
-                    it.copy(rawResult = null)
-                }
-            } catch (e: ScanException) {
-                e.showStackTrace()
-
-                log.error { "Could not scan '${pkg.id.toCoordinates()}': ${e.collectMessagesAsString()}" }
-
-                val now = Instant.now()
-                listOf(ScanResult(
-                        provenance = Provenance(),
-                        scanner = scannerDetails,
-                        summary = ScanSummary(
-                                startTime = now,
-                                endTime = now,
-                                fileCount = 0,
-                                licenseFindings = sortedSetOf(),
-                                errors = listOf(OrtIssue(source = scannerName, message = e.collectMessagesAsString()))
-                        ),
-                        rawResult = EMPTY_JSON_NODE)
-                )
+        val cacheDispatcher = newFixedThreadPoolContext(5, "cache")
```

Comment:
Should we replace "cache" with "storage" here?

Reply:
Done.